src/inmanta/loader.py

```diff
@@ -397,66 +395,138 @@ def convert_module_to_relative_path(cls, full_mod_name: str) -> str:
-            if os.path.isdir(path_to_module):
-                path_to_module = os.path.join(path_to_module, "__init__.py")
-                if os.path.exists(path_to_module):
-                    return path_to_module
-
-        raise ImportError(f"Cannot find module {fullname} in {self._modulepaths}")
-
     def _loading_top_level_package(self):
         return self.path == ""
 
 
 class PluginModuleFinder(Finder):
     """
-    Custom module finder which handles all the imports for the package inmanta_plugins.
+    Custom module finder which handles V1 Inmanta modules. V2 modules are handled using the standard Python finder.
+
+    This finder is stored as the fist entry in `meta_path`, before the default Python finders. As such, this finder should be
+    configured to ignore certain modules when these modules are V2 modules.
     """
 
-    def __init__(self, modulepaths: List[str]) -> None:
-        self._modulepaths = modulepaths
+    MODULE_FINDER: "PluginModuleFinder" = None
+
+    def __init__(self, modulepaths: List[str], modules_to_ignore: List[str] = []) -> None:
+        """
+        :param modulepaths: The module paths for the inmanta project.
+        :param modules_to_ignore: The modules that should be ignored by this finder.
+        """
+        self._modulepaths = list(modulepaths)
+        self._modules_to_ignore = list(modules_to_ignore)
+
+    @classmethod
+    def get_module_finder(cls) -> "PluginModuleFinder":
+        if cls.MODULE_FINDER is not None:
+            return cls.MODULE_FINDER
+        raise Exception("No PluginModuleFinder configured. Call configure_module_finder() first.")
+
+    @classmethod
+    def reset(cls) -> None:
+        """
+        Remove the PluginModuleFinder from sys.meta_path and unload the inmanta_plugins package.
+        """
+        if cls.MODULE_FINDER is not None and cls.MODULE_FINDER in sys.meta_path:
+            sys.meta_path.remove(cls.MODULE_FINDER)
+        unload_inmanta_plugins()
+        cls.MODULE_FINDER = None
+
+    @classmethod
+    def configure_module_finder(cls, modulepaths: List[str], modules_to_ignore: List[str] = []) -> None:
+        """
+        Setup a custom module loader to handle imports in .py files of the modules. This finder will be stored
+        as the first finder in sys.meta_path.
+
+        :param modulepaths: The directories where the module finder should look for modules.
+        :param modules_to_ignore: The module that should not be handled by the module finder.
+        """
+        if cls.MODULE_FINDER is not None:
+            # PluginModuleFinder already present in sys.meta_path
+            cls.MODULE_FINDER.add_module_paths(modulepaths)
+            cls.MODULE_FINDER._modules_to_ignore = list(modules_to_ignore)
+            return
+
+        # PluginModuleFinder not yet present in sys.meta_path.
+        module_finder = PluginModuleFinder(modulepaths, modules_to_ignore)
+        sys.meta_path.insert(0, module_finder)
```

Comment:
Why do we give this loader precedence if we want to have V2 module have precedence?

Reply:
That's indeed strange. I will fix it.

Reply:
No. This will change to `sys.meta_path.append(module_finder)` instead.

Reply:
I mean the docstring contains"This finder will be stored as the first finder in sys.meta_path".